crates/wasmtime/src/runtime/component/func/host.rs

```diff
@@ -819,6 +853,10 @@ where
             );
         }
     } else {
+        if async_function {
+            concurrent::check_blocking(store.0)?;
+        }
```

Comment:
Can you leave a comment on checks like this to indicate what the trap/check represents?